bin/mail/2017-01-19-new-york-PD/generate-list

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env ruby
+require_relative '../mailing-common/mailing-list-utils'
+
+# This is using the same main queries as 2017-01-05-go-beyond-an-hour,
+# filtered to ONLY @schools.nyc.gov, @strongschools.nyc emails (previously those were excluded).
+
+# Use reporting DBs
+PEGASUS_REPORTING_DB = sequel_connect CDO.pegasus_reporting_db_writer, CDO.pegasus_reporting_db_reader
+DASHBOARD_REPORTING_DB = sequel_connect CDO.dashboard_reporting_db_writer, CDO.dashboard_reporting_db_reader
+
+# 3 categories:
+CATEGORIES = [
+  # 1) us_hoc_no_account - US 2016 Hour of Code organizers who do not have Code Studio accounts
+  CATEGORY_US_HOC = 'us_hoc_no_account'.freeze,
+
+  # 2) international_hoc_no_account - international 2016 Hour of Code organizers who do not have Code Studio accounts
+  CATEGORY_INTERNATIONAL_HOC = 'international_hoc_no_account'.freeze,
+
+  # 3) teachers_no_pd - US Teachers with Code Studio accounts that have not gone through our PD
+  CATEGORY_TEACHERS_NO_PD = 'teachers_no_pd'.freeze
+].freeze
+
+# In all cases, filter to @schools.nyc.gov and @strongschools.nyc
+
+@results = {}
+
+def create_temp_teachers_table
+  PEGASUS_REPORTING_DB.run "
+    CREATE TEMPORARY TABLE IF NOT EXISTS
+      temp_teachers (INDEX(email))
+    AS (
+      SELECT id, email COLLATE utf8_general_ci AS email
+      FROM dashboard_production.users
+      WHERE user_type = 'teacher'
+    );
+  "
+end
+
+# 1) us_hoc_no_account - US 2016 Hour of Code organizers who do not have Code Studio accounts
+# and
+# 2) international_hoc_no_account - international 2016 Hour of Code organizers who do not have Code Studio accounts
+def query_hoc_no_account
+  create_temp_teachers_table
+
+  pegasus_query = "
+    SELECT forms.email, forms.name, forms.data
+    FROM forms
+    LEFT OUTER JOIN temp_teachers AS teachers
+      ON forms.email = teachers.email
+    WHERE
+      forms.kind = 'HocSignup2016'
+      AND teachers.id IS NULL
+      AND (
+        forms.email LIKE '%@schools.nyc.gov'
+        OR forms.email LIKE '%@strongschools.nyc'
+      )
+    ORDER BY forms.id DESC
+  "
+
+  PEGASUS_REPORTING_DB.fetch(pegasus_query).each do |row|
+    email = row[:email]
+    next if UNSUBSCRIBERS.include? email
+
+    data = JSON.parse(row[:data])
+    hoc_country = data['hoc_country_s']
+    category = hoc_country == 'us' ? CATEGORY_US_HOC : CATEGORY_INTERNATIONAL_HOC
```

Comment:
no. Fortunately, `hoc_country_s` is from a set [list](https://github.com/code-dot-org/code-dot-org/blob/cf2f4da5e0d894c113e5fc47ef03f0138e3b0ca3/pegasus/sites.v3/hourofcode.com/i18n/countries.json)